### PR TITLE
MGMT-3420 None platform additional changes

### PR DESCRIPTION
**Pull request by ori-amizur**
None platform additional changes
- Add ports 443 and 80 to libvirt firewall for none platform
- Make DNS queries local in terraform
- Make load balancer access workers for ports 80 and 443
- Add NAT iptables rules for none platform IPv4 since it is using route forwarding mode
@YuviGold 
@tsorya

### CODE REVIEW

**Review by YuviGold — APPROVED** [review]
LGTM

**Comment by YuviGold** [comment]
Please update the commit message with all the relevant additions

File: Makefile [hunk 1]
```diff
@@ -167,6 +167,14 @@ start_load_balancer:
 stop_load_balancer:
 	id=`podman ps --quiet --filter "name=load_balancer"`; test ! -z "$$id"  && podman rm -f load_balancer ; rm -f  $(HOME)/.test-infra/etc/nginx/stream.d/*.conf >& /dev/null || /bin/true
 
+########
+# Nat  #
+########
+
+stop_nat:
```
Comment (tsorya) [inline Makefile:174]: can we have some explanation ? it looks like magic

Comment (ori-amizur) [reply Makefile:174]: It removes all iptables nat rules having the mark 555.  Look at nat_controller.py

File: discovery-infra/test_infra/controllers/nat_controller.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    def _insert_rule(self, rule_suffix):
+        insert_rule = self._build_rule_string('insert', rule_suffix)
+        logging.info("Adding rule \"%s\"", insert_rule)
+        run_command(insert_rule, shell=True)
+
+    def _add_rule(self, rule_suffix):
+        if not self._does_rule_exist(rule_suffix):
+            self._insert_rule(rule_suffix)
+
+    def add_nat_rules(self, input_interfaces):
```
Comment (tsorya) [inline discovery-infra/test_infra/controllers/nat_controller.py:39]: will this work in e2e test in parallel mode?

Comment (tsorya) [reply discovery-infra/test_infra/controllers/nat_controller.py:39]: maybe we should delete created rules too?

Comment (ori-amizur) [reply discovery-infra/test_infra/controllers/nat_controller.py:39]: It will work in parallel mode

Comment (ori-amizur) [reply discovery-infra/test_infra/controllers/nat_controller.py:39]: @tsorya where can we delete them? Currently I added a target that removes them in Makefile destroy target

Comment (tsorya) [reply discovery-infra/test_infra/controllers/nat_controller.py:53]: i mean in e2e? we will need to delete them right? or all none platform tests shouldn't be run in parallel ?

Comment (ori-amizur) [reply discovery-infra/test_infra/controllers/nat_controller.py:53]: The nat has two parts: The first is handle the input interfaces which is per cluster.  The second is the output interface(s) which is on host level.  I can try to remove the handling on cluster level.  destroy will remove all.

File: discovery-infra/start_discovery.py [hunk 3]
```diff
@@ -434,9 +442,12 @@ def nodes_flow(client, cluster_name, cluster):
 
         if is_none_platform_mode() and args.master_count > 1:
             master_ips = helper_cluster.Cluster.get_master_ips(client, cluster.id, main_cidr) + helper_cluster.Cluster.get_master_ips(client, cluster.id, secondary_cidr)
+            worker_ips = helper_cluster.Cluster.get_worker_ips(client, cluster.id, main_cidr) + helper_cluster.Cluster.get_worker_ips(client, cluster.id, secondary_cidr)
+            if not worker_ips:
+                worker_ips = master_ips
             load_balancer_ip = _get_host_ip_from_cidr(machine_net.cidr_v6 if machine_net.has_ip_v6 and not machine_net.has_ip_v4 else machine_net.cidr_v4)
             lb_controller = LoadBalancerController(tf)
-            lb_controller.set_load_balancing_config(load_balancer_ip, master_ips)
+            lb_controller.set_load_balancing_config(load_balancer_ip, master_ips, worker_ips)
```
Comment (YuviGold) [inline discovery-infra/start_discovery.py:450]:
```python
            lb_controller.set_load_balancing_config(load_balancer_ip, master_ips, worker_ips if worker_ips else master_ips)
```

File: discovery-infra/test_infra/controllers/load_balancer_controller.py [hunk 4]
```diff
@@ -28,8 +28,9 @@ def _render_port_entities(self, load_balancer_ip, master_ips, port):
         upstream_name = f'upstream_{re.sub(r"[.:]", r"_", load_balancer_ip)}_{port}'
         return self._render_upstream_block(master_ips, port, upstream_name) + self._render_server_block(load_balancer_ip, port, upstream_name)
 
-    def _render_load_balancer_config_file(self, load_balancer_ip, master_ips):
-        return '\n'.join([self._render_port_entities(load_balancer_ip, master_ips, port) for port in [443, 6443, 22623]])
+    def _render_load_balancer_config_file(self, load_balancer_ip, master_ips, worker_ips):
+        return '\n'.join([self._render_port_entities(load_balancer_ip, master_ips, port) for port in [6443, 22623]]) + '\n' + \
```
Comment (YuviGold) [inline discovery-infra/test_infra/controllers/load_balancer_controller.py:32]: ignition port

**Pull request merged**